### PR TITLE
Added LogStep override and made sure that FinishTest/Step doesn't try…

### PR DESCRIPTION
**Pull request by nafonso**
… to be called multiple times in a row

#### Description
Change for ASpatialFunctionalTests to automatically fail the test if there's a failed assert.

### CODE REVIEW

**Comment by nafonso** [comment]
Engine PRs to make this one compile:
4.24: https://github.com/improbableio/UnrealEngine/pull/632
4.25: https://github.com/improbableio/UnrealEngine/pull/633

File: SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp [hunk 1]
```diff
@@ -112,6 +112,16 @@ void ASpatialFunctionalTest::RegisterAutoDestroyActor(AActor* ActorToAutoDestroy
 	}
 }
 
+void ASpatialFunctionalTest::LogStep(ELogVerbosity::Type Verbosity, const FString& Message)
+{
+	Super::LogStep(Verbosity, Message);
+
+	if(Verbosity == ELogVerbosity::Error || Verbosity == ELogVerbosity::Fatal)
+	{
+		FinishTest(EFunctionalTestResult::Failed, TEXT("Failed assertions"));
```
Comment (mironec) [inline SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:121]: Maybe we should also set the current step index to something invalid - or is that already handled on FinishTest?

Comment (nafonso) [reply SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTest.cpp:121]: That's handled by the Test Authority as a flow to do the cleanup process as well.

**Review by mironec — APPROVED** [review]
May also need the spatial version increase.

**Comment by nafonso** [comment]
> May also need the spatial version increase.

Done.

**Pull request merged**